frontend/sozisel-app/src/components/PresenterSession/EventsTimline/EventsTimeline.tsx

```diff
@@ -0,0 +1,220 @@
+import "./EventsTimeline.scss";
+import "moment-timezone";
+
+import * as moment from "moment";
+
+import {
+  Button,
+  IconButton,
+  Paper,
+  Snackbar,
+  Step,
+  StepLabel,
+  Stepper,
+  Typography,
+} from "@material-ui/core";
+import React, { ReactElement, useEffect, useState } from "react";
+import {
+  useEndSessionMutation,
+  useLaunchEventMutation,
+} from "../../../graphql";
+
+import { AUTO_HIDE_DURATION } from "../../../common/consts";
+import { Alert } from "@material-ui/lab";
+import CalendarTodayIcon from "@material-ui/icons/CalendarToday";
+import { Event } from "../../../model/Template";
+import { Participant } from "../../../hooks/useLiveSessionParticipation";
+import PlayCircleFilledIcon from "@material-ui/icons/PlayCircleFilled";
+import QuizDetails from "./QuizDetails/QuizDetails";
+import useCountdownTimer from "../../../hooks/useCountdownTimer";
+import { useHistory } from "react-router";
+import { useTranslation } from "react-i18next";
+
+interface LaunchedEvent {
+  id: string;
+  insertedAt: string;
+}
+
+interface EventsTimelineProps {
+  events: Event[];
+  launchedEvents: LaunchedEvent[];
+  sessionId: string;
+  participants: Participant[];
+}
+
+interface TimerProps {
+  onFinishCallback: () => void;
+  startValue: number;
+}
+
+const Timer = ({ onFinishCallback, startValue }: TimerProps): ReactElement => {
+  const countdownTimer = useCountdownTimer({
+    startValue,
+    onFinishCallback,
+  });
+
+  return <p className="Timer">{countdownTimer}</p>;
+};
+
+const getRandomParticipants = (
+  participants: Participant[],
+  targetPercentageOfParticipants: number
+): string[] => {
+  return participants
+    .sort(() => 0.5 - Math.random())
+    .slice(
+      0,
+      Math.floor((targetPercentageOfParticipants / 100) * participants.length)
+    )
+    .map((participant) => participant.id);
+};
+
+export default function EventsTimeline({
+  events,
+  launchedEvents,
+  sessionId,
+  participants,
+}: EventsTimelineProps): ReactElement {
+  const { t } = useTranslation("common");
+  const [activeEventIdx, setActiveEventIdx] = useState<number>(0);
+  const [activeEventId, setActiveEventId] = useState<string>("");
+  const [activeEventCurrentSec, setActiveEventCurrentSec] = useState<number>(0);
+  const history = useHistory();
+
+  const [endSessionMutation, { error: endSessionError }] =
+    useEndSessionMutation();
+
+  const onEndSession = async () => {
+    await endSessionMutation({ variables: { id: sessionId } });
```

Comment:
You know that when you get an error on await the function will throw and you would not get a `endSessionError`? Do the try/catch and the error in catch statement will be the eventual mutation error.

Reply:
Hmmm, I tested it, and it works properly. 
![image](https://user-images.githubusercontent.com/47144579/124805530-edd35480-df5b-11eb-858f-e18301d93b6a.png)
![image](https://user-images.githubusercontent.com/47144579/124805643-122f3100-df5c-11eb-84cb-2dfaaeae4d0c.png)
Or I'm stupid and I don't understand you.

Reply:
This log is done by uncaught exception and not your log... The error is set asynchronously but not in this place

Reply:
done.